exec/java-exec/src/main/java/org/apache/drill/exec/physical/rowSet/RowSetStringBuilder.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.physical.rowSet;
+
+import org.apache.drill.exec.record.BatchSchema.SelectionVectorMode;
+import org.apache.drill.exec.record.metadata.ColumnMetadata;
+import org.apache.drill.exec.record.metadata.TupleMetadata;
+
+/**
+ * Helper class to obtain string representation of RowSet.
+ * Example of the output:
+ * <pre>
+ *   #: year, month, day
+ *   0: 2017, 12, 17
+ *   1: 2017, 12, 18
+ *   2: 2017, 12, 19
+ * </pre>
+ */
+public class RowSetStringBuilder {
```

Comment:
The problem with this approach is that row sets can be large: creating a string to print causes memory pressure.

Consider creating a row set builder that takes a stream (or Writer). The Writer can be a String writer. That way, when used with a large data set, the writer can be to stdout, or whatever.

Reply:
The class name is still "RowSetStringBuilder" Better to rename it to, say, "RowSetFormatter". The constructor should take both a row set and a writer. We can then provide static methods for the two most common cases: print to stdout (which I use all the time when creating tests) and to string (which you seem to prefer):

```
public static void print(RowSet rowSet) {
  new RowSetFormatter(rowSet, System.out).write();
}

public static String toString(RowSet rowSet) {
  StringWriter out = new StringWriter();
  new RowSetFormatter(rowSet, out).write();
  return out.toString();
}
```

The constructor allows other options, such as writing to a file. Sadly, it won't work to write to a log; we'd have to materialize to a string...